domain/service/servicestore.go

```diff
@@ -82,47 +105,81 @@ func (s *storeImpl) Put(ctx datastore.Context, svc *Service) error {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("storeImpl.Put"))
 	svc.ConfigFiles = make(map[string]servicedefinition.ConfigFile)
 
-	return s.ds.Put(ctx, Key(svc.ID), svc)
+	err := s.ds.Put(ctx, Key(svc.ID), svc)
+	if err == nil {
+		s.updateVolatileInfo(svc.ID, svc.DesiredState, svc.UpdatedAt) // Mutex Lock
+	}
+	return err
+}
+
+// UpdateDesiredState updates the DesiredState for the service by saving the information in volatile storage.
+func (s *storeImpl) UpdateDesiredState(ctx datastore.Context, serviceID string, desiredState int) error {
+	plog.WithFields(log.Fields{
+		"serviceID": serviceID,
+		"desiredState": desiredState,
+	}).Debug("Storing desiredState")
+	s.updateVolatileInfo(serviceID, desiredState, time.Now())
+	return nil
 }
 
 // Get a Service by id. Return ErrNoSuchEntity if not found
 func (s *storeImpl) Get(ctx datastore.Context, id string) (*Service, error) {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("storeImpl.Get"))
+	// Get the service from elastic and fill additional info into the
+	// service object.
+	svc, err := s.get(ctx, id)
+	if err == nil {
```

Comment:
Please remove lines 131-133; they are not necessary because of the call to fillAdditionalInfo at line 144